functions/slackEvents.js

```diff
@@ -68,3 +137,81 @@ async function onMessageEvent(event) {
     text: `<@${event.user}> commented on a thread in <#${event.channel}>: _<${linkRes.permalink}|link to comment>_`,
   });
 }
+
+async function handleAction(action, user) {
+  switch (action.action_id) {
+    case 'email_shopping_list':
+      try {
+        await emailShoppingList(action.value, user.id);
+      } catch (err) {
+        console.error('Failed to email shopping list to user', { user: user.id, err: err });
+        return 1;
+      }
+
+      return 0;
+    default:
+      console.error('Encountered an unsupported slack action_id', { action_id: action.action_id });
+      return 1;
+  }
+}
+
+async function sendMessageToSlackResponseUrl(responseUrl, message) {
+  console.log(JSON.stringify(message));
+  return await fetch(responseUrl, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(message)
+  });
+}
+
+async function emailShoppingList(ticketID, userID) {
+  const records = await getRecordsWithTicketID(INTAKE_TABLE, ticketID);
+  if (records.length !== 1) {
+    throw new Error(`Found ${records.length} records`);
+  }
+
+  const [, volunteer] = await getVolunteerBySlackID(userID);
+  if (!volunteer) {
+    throw new Error(`No volunteer found for slack id ${userID}`);
+  }
+
+  const shoppingList = await getShoppingList(records);
+
+  const email = `
+Hi ${volunteer.Name.split(' ')[0]}!
+
+Thank you for volunteering to deliver groceries to our neighbors with Bed-Stuy Strong!
+
+Here's your shopping list for ticket ${ticketID}:
```

Comment:
I'd like to put more ticket metadata in here (name, address, phone number, vulnerabilities, household size, etc) so I can copy the whole thing into my phone in one go). But again, can do those things post merge.